File: lib/models/ride.dart
class Ride {
  final String serviceProvider; // Ola, Uber, or Rapido
  final double price;
  final String vehicleType; // Mini, Prime, Bike, etc.
  final int estimatedTimeInMinutes;
  final double distance;
  final String? trackingUrl;
  final Map<String, dynamic>? additionalInfo;

  Ride({
    required this.serviceProvider,
    required this.price,
    required this.vehicleType,
    required this.estimatedTimeInMinutes,
    required this.distance,
    this.trackingUrl,
    this.additionalInfo,
  });

  factory Ride.fromJson(Map<String, dynamic> json) {
    return Ride(
      serviceProvider: json['serviceProvider'] as String,
      price: (json['price'] as num).toDouble(),
      vehicleType: json['vehicleType'] as String,
      estimatedTimeInMinutes: json['estimatedTimeInMinutes'] as int,
      distance: (json['distance'] as num).toDouble(),
      trackingUrl: json['trackingUrl'] as String?,
      additionalInfo: json['additionalInfo'] as Map<String, dynamic>?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'serviceProvider': serviceProvider,
      'price': price,
      'vehicleType': vehicleType,
      'estimatedTimeInMinutes': estimatedTimeInMinutes,
      'distance': distance,
      'trackingUrl': trackingUrl,
      'additionalInfo': additionalInfo,
    };
  }
} 

File: backend/app/models/ride.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any

class RideResponse(BaseModel):
    service_provider: str = Field(..., description="Name of the service provider (Ola, Uber, Rapido)")
    price: float = Field(..., description="Ride price in INR")
    vehicle_type: str = Field(..., description="Type of vehicle (Mini, Prime, Bike, etc.)")
    estimated_time_minutes: int = Field(..., description="Estimated time in minutes")
    distance: float = Field(..., description="Distance in kilometers")
    tracking_url: Optional[str] = Field(None, description="URL for live tracking")
    additional_info: Optional[Dict[str, Any]] = Field(None, description="Additional provider-specific information")

    class Config:
        json_schema_extra = {
            "example": {
                "service_provider": "Uber",
                "price": 250.0,
                "vehicle_type": "UberGo",
                "estimated_time_minutes": 15,
                "distance": 8.5,
                "tracking_url": "https://track.uber.com/xyz",
                "additional_info": {
                    "surge_multiplier": 1.2,
                    "driver_rating": 4.5
                }
            }
        } 